skimage/future/graph/rag.py

```diff
@@ -269,30 +278,6 @@ def _add_node_silent(self, n):
         .. seealso:: :func:`networkx.Graph.add_node`."""
         super(RAG, self).add_node(n)
 
-    def nodes_iter(self, *args, **kwargs):
-        """ Iterate over nodes
-
-        For compatibility with older versions of networkx.  Versions <= 1.11
```

Comment:
We need to bump the dependency to version `> 1.11` then (https://github.com/scikit-image/scikit-image/blob/master/requirements.txt).

Reply:
As I recall there was a reason the minimum version of networkx was where it was - had to do with our other minimum dependencies.  Networkx has been more aggressive about shedding legacy support than we have.  We should double check that this bump doesn't unduly change the rest of the compatibility matrix.

Reply:
``nodes_iter`` and ``edges_iter`` are provided by both 1.x and 2.x now.

Reply:
We spent the last week making sure v2.x was sufficiently backwards compatible while working on this PR, so there is no reason to bump your dependencies now.  At some point in the future, we will release v3.x.  At that point you will need to drop support for v1.x.  But with the changes in this PR, your code should run fine on v1.x and v2.x.

If you want to see the changes we made to do this, here they are:
- https://github.com/networkx/networkx/pull/2635
- https://github.com/networkx/networkx/pull/2633
- https://github.com/networkx/networkx/pull/2636
- https://github.com/networkx/networkx/pull/2639

All but the last networkx PR were directly a result of working on this scikit-image PR.

Reply:
Just to be clear about ```nodes_iter``` and ```edges_iter```.  These methods do not exist in the NXv2 graph classes. But code written for both NXv1 and NXv2 should not use these methods, just as we don't use ```dict.iterkeys()``` in code written for both python2 and python3. So these methods should not be defined in RAG if we envision RAG as working for both v2 and v1.

Leaving them out will mean that anyone using RAG with NXv1 will have those methods available (inherited from the base class). Anyone using RAG with NXv2 will not have those methods available. Anyone writing code that might work for NXv1 or NXv2 should not use these methods.

Reply:
Thanks for the correction.